src/main/java/net/sf/jabref/importer/fetcher/IEEEXploreFetcher.java

```diff
@@ -125,71 +95,137 @@ public JPanel getOptionsPanel() {
+
+            //if no search results were found
+            if (hits == 0) {
+                status.showMessage(Localization.lang("No entries found for the search string '%0'", query),
                         DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
                 return false;
             }
 
-            if (page.contains("Error Page")) {
+            //if max hits were exceeded, display the warning
+            if (hits > IEEEXploreFetcher.MAX_FETCH) {
                 // @formatter:off
-                status.showMessage(
-                        Localization.lang("Intermittent errors on the IEEE Xplore server. Please try again in a while."),
-                       DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
+                status.showMessage(Localization.lang("%0 entries found. To reduce server load, only %1 will be downloaded.",
+                        new String[] {String.valueOf(hits), String.valueOf(IEEEXploreFetcher.MAX_FETCH)}),
+                        DIALOG_TITLE, JOptionPane.INFORMATION_MESSAGE);
                 // @formatter:on
-                return false;
             }
 
-            hits = getNumberOfHits(page, "display-status", hitsPattern);
+            //buffer to use for building the URL for fetching the bibtex data from IEEEXplore
+            StringBuffer bibtexQueryURLStringBuf = new StringBuffer();
+            bibtexQueryURLStringBuf.append(URL_BIBTEX_START);
 
+            //loop over each record and create a comma-separate list of article numbers which will be used to download the raw Bibtex
+            JSONArray recordsJsonArray = jsonObj.getJSONArray("records");
+            for (int n = 0; n < recordsJsonArray.length(); n++) {
+                if (!recordsJsonArray.getJSONObject(n).isNull("articleNumber")) {
+                    bibtexQueryURLStringBuf.append(recordsJsonArray.getJSONObject(n).getString("articleNumber") + ",");
+                }
+            }
+            //delete the last comma
+            bibtexQueryURLStringBuf.deleteCharAt(bibtexQueryURLStringBuf.length() - 1);
+
+
+            //add the abstract setting
             includeAbstract = absCheckBox.isSelected();
-            if (hits > IEEEXploreFetcher.MAX_FETCH) {
-                // @formatter:off
-                status.showMessage(Localization.lang("%0 entries found. To reduce server load, only %1 will be downloaded.",
-                        new String[] {String.valueOf(hits), String.valueOf(IEEEXploreFetcher.MAX_FETCH)}),
+            if (includeAbstract) {
+                bibtexQueryURLStringBuf.append("&citations-format=citation-abstract");
+            } else {
+                bibtexQueryURLStringBuf.append("&citations-format=citation-only");
+            }
+
+            //append the remaining URL
+            bibtexQueryURLStringBuf.append(URL_BIBTEX_END);
+
+            //fetch the raw Bibtex results from IEEEXplore and parse
+            URL bibtexURL = new URL(bibtexQueryURLStringBuf.toString());
```

Comment:
This method is really large. Can you split it up a little bit? May be creating a method to compute the bibtexQueryURL can help.